app/domain/authentication/authn_jwt/fetch_issuer_value.rb

```diff
@@ -0,0 +1,131 @@
+require 'uri'
+
+module Authentication
+  module AuthnJwt
+
+    # FetchIssuerValue command class is responsible to fetch the issuer secret value,
+    # in order to validate it later against the JWT token claim
+    FetchIssuerValue ||= CommandClass.new(
+      dependencies: {
+        resource_class: ::Resource,
+        fetch_secrets: ::Conjur::FetchRequiredSecrets.new,
```

Comment:
can we use `FetchAuthenticatorSecrets` instead? please see reference here: https://github.com/cyberark/conjur/blob/31a3603fca360724868b9ab61fa7cc7474a0956a/app/domain/authentication/authn_azure/authenticator.rb#L8

Reply:
it can save you some logic that you implement in this class (build resource id, the double memoization in `issuer_secret`, etc.)

Reply:
The use  of `FetchAuthenticatorSecrets` command class only encapsulate the `resource_id` function, but here i need this logic anyway for the `Resource` class , to verify if resources are exist before i am fetching their secrets.

So in  this case i think that i should use the `FetchRequiredSecrets` command class which get as input the resource id

Reply:
i would still use `FetchAuthenticatorSecrets`. you are already masking the `ResourceNotFound` error so the code will be cleaner. you will also go to the db once instead of 3 times.

Is it possible? even it means to make some change to the FetchSecrets class.

Not a blocker.

Reply:
I dont understand why i will go to the db once instead of 3 times.
Anyway i will take the "Not a blocker" ticket :)